hail/python/hailtop/aiogoogle/client/compute_client.py

```diff
@@ -77,13 +82,50 @@ async def list(self, path: str, *, params: MutableMapping[str, Any] = None, **kw
         return PagedIterator(self, path, params, kwargs)
 
     async def create_disk(self, path: str, *, params: MutableMapping[str, Any] = None, **kwargs):
-        return await request_with_wait_for_done(self.post, path, params, **kwargs)
+        return await self._request_with_zonal_operations_response(self.post, path, params, **kwargs)
 
     async def attach_disk(self, path: str, *, params: MutableMapping[str, Any] = None, **kwargs):
-        return await request_with_wait_for_done(self.post, path, params, **kwargs)
+        return await self._request_with_zonal_operations_response(self.post, path, params, **kwargs)
 
     async def detach_disk(self, path: str, *, params: MutableMapping[str, Any] = None, **kwargs):
-        return await request_with_wait_for_done(self.post, path, params, **kwargs)
+        return await self._request_with_zonal_operations_response(self.post, path, params, **kwargs)
 
     async def delete_disk(self, path: str, *, params: MutableMapping[str, Any] = None, **kwargs):
-        return await request_with_wait_for_done(self.delete, path, params, **kwargs)
+        return await self.delete(path, params=params, **kwargs)
+
+    async def _request_with_zonal_operations_response(self, request_f, path, params: MutableMapping[str, Any] = None, **kwargs):
+        assert 'params' not in kwargs
+
+        async def request_and_wait():
+            local_params = copy.deepcopy(params)
+
+            if local_params is None:
+                local_params = {}
+
+            request_uuid = str(uuid.uuid4())
+            if 'requestId' not in local_params:
+                local_params['requestId'] = request_uuid
+
+            resp = await request_f(path, params=local_params, **kwargs)
```

Comment:
Hmm. I'm wary of deep copying params.

Do you agree that it's always a bug for a user of this API to set `requestId`? If they set a requestId, there's no way for us to correctly retry transient errors.

In light of that, I think we should do this:

```python3
async def _request_with_zonal_operations_response(...):
    params = params or dict()
    assert 'requestId' not in params
    async def request_and_wait():
        params['requestId'] = str(uuid.uuid4())
        resp = ...
```